src/Repository/DonationRepository.php

```diff
@@ -47,23 +47,22 @@ public function findAllSubscribedDonationByEmail(string $email): array
                 'email' => $email,
                 'duration' => PayboxPaymentSubscription::NONE,
             ])
-            ->orderBy('donation.donatedAt', 'ASC')
+            ->orderBy('donation.donatedAt', 'DESC')
             ->getQuery()
             ->getResult()
         ;
     }
 
-    public function findLastSubscribedDonationByEmail(string $email): ?Donation
+    public function findLastSubscriptionEndedDonationByEmail(string $email): ?Donation
     {
         return $this->createQueryBuilder('donation')
             ->andWhere('donation.emailAddress = :email')
-            ->andWhere('donation.duration != :duration')
             ->andWhere('donation.subscriptionEndedAt IS NOT NULL')
             ->setParameters([
                 'email' => $email,
                 'duration' => PayboxPaymentSubscription::NONE,
             ])
-            ->orderBy('donation.donatedAt', 'ASC')
+            ->orderBy('donation.donatedAt', 'DESC')
```

Comment:
It should be ordered by subscriptionEndedAt no ?

Reply:
Oooo yes, my brain not started yet ^^,